oak/server/rust/oak_runtime/src/introspection_events.rs

```diff
@@ -44,9 +43,10 @@ impl Runtime {
             event_details: Some(event_details),
         };
 
-        // TODO(#913): Push the event over to a channel, to be consumed by the
-        // introspection aux server.
-        debug!("Introspection event recorded: {:?}", event);
+        self.introspection_event_queue
+            .lock()
+            .unwrap()
+            .push_back(event);
```

Comment:
Is it bad idea to call `lock` from within the Runtime? If the introspection aux server were to currently hold the lock for that `Mutex` it seems we'd block the Runtime, which seems troublesome. 

I could spawn a thread to mitigate this.

Reply:
Are you expecting the introspection aux server to hold the lock for a long period (e.g. while a user is scrubbing backwards and forwards through the events)?  If so, then you're right we'd need something to prevent it blocking the Runtime.

OTOH if the introspection code is only going to grab the lock momentarily> then it should be OK.  (However, that might in turn assume that the ordering of events in the `VecDeque` is stable, e.g. that it's OK to lock/read-entry-17/unlock  then lock/read-entry-16/unlock and have the numbering not change underneath.)

Reply:
For now it should _probably_ be fine. There's a point to be made that responding to scrubbing should probably be done client side, in which case past events would be serialized to JSON upon requesting `/graph`. Hence the lock should be temporary. 

For now we should be ok with ordering, since the order of the queue is of infinite length & chronological. However admittedly that might change down the road.